tests/lib/Token/RandomBytesGeneratorTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Core\Token;
+
+use Ibexa\Contracts\Core\Token\TokenGeneratorInterface;
+use Ibexa\Core\Token\RandomBytesGenerator;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Ibexa\Core\Token\RandomBytesGenerator
+ */
+final class RandomBytesGeneratorTest extends TestCase
+{
+    private TokenGeneratorInterface $tokenGenerator;
+
+    protected function setUp(): void
+    {
+        $this->tokenGenerator = new RandomBytesGenerator();
+    }
+
+    /**
+     * @dataProvider provideDataForTestGenerateToken
+     *
+     * @throws \Exception
+     */
+    public function testGenerateToken(
+        int $expectedTokenLengthInBytes,
+        int $length
+    ): void {
+        $generatedToken = $this->tokenGenerator->generateToken($length);
+
+        // Check if Generator returns different tokens
+        self::assertNotSame(
+            $generatedToken,
+            $this->tokenGenerator->generateToken($length)
+        );
```

Comment:
Use assertion message instead.

```suggestion
        self::assertNotSame(
            $generatedToken,
            $this->tokenGenerator->generateToken($length),
            'Token generator should return different values on subsequent calls',
        );
```